node/src/components/gossiper/gossip_table.rs

```diff
@@ -209,25 +209,25 @@ impl<T: Copy + Eq + Hash + Display> GossipTable<T> {
             return GossipAction::Noop;
         }
 
-        if let Some(state) = self.paused.get_mut(data_id) {
-            let _ = state.holders.insert(holder);
-            return GossipAction::Noop;
+        let update = |state: &mut State| {
+            let _ = state.holders.insert(holder.clone());
+        };
```

Comment:
Since `state` will come from the methods that we pass the `update` to, couldn't we also pass in the `holder` and get rid of this lambda?

Reply:
It's more like we're doing different updates depending upon where the call is being made, so we need to pass the closure to the `self.update_current()` or `self.update_paused()` since we're not always making the same changes to the state found in `current` or `paused`.

Reply:
NVM, I read the rest of the code.